db/views/reporting_patient_states_per_month_v03.sql

```diff
@@ -0,0 +1,173 @@
+    visits.appointment_recorded_at AS appointment_recorded_at,
+
+    ------------------------------------------------------------
+    -- relative time calculations
+
+    (cal.year - DATE_PART('year', p.recorded_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE')))) * 12 +
+    (cal.month - DATE_PART('month', p.recorded_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE'))))
+    AS months_since_registration,
+
+    (cal.year - DATE_PART('year', p.recorded_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE')))) * 4 +
+    (cal.month - DATE_PART('quarter', p.recorded_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE'))))
+    AS quarters_since_registration,
+
+    visits.months_since_visit AS months_since_visit,
+    visits.quarters_since_visit AS quarters_since_visit,
+    bps.months_since_bp AS months_since_bp,
+    bps.quarters_since_bp AS quarters_since_bp,
+
+    ------------------------------------------------------------
+    -- indicators
+    CASE
+        WHEN (bps.systolic IS NULL OR bps.diastolic IS NULL) THEN 'unknown'
+        WHEN (bps.systolic < 140 AND bps.diastolic < 90) THEN 'controlled'
+        ELSE 'uncontrolled'
+        END
+        AS last_bp_state,
+
+    CASE
+        WHEN p.status = 'dead' THEN 'dead'
+        WHEN (
+          -- months_since_registration
+          (cal.year - DATE_PART('year', p.recorded_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE')))) * 12 +
+          (cal.month - DATE_PART('month', p.recorded_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE')))) < 12
+
+          OR
+
+          bps.months_since_bp < 12
+        ) THEN 'under_care'
+        ELSE 'lost_to_follow_up'
+        END
+        AS htn_care_state,
+
+    CASE
+        WHEN (visits.months_since_visit >= 3 OR visits.months_since_visit is NULL) THEN 'missed_visit'
+        WHEN (bps.months_since_bp >= 3 OR bps.months_since_bp is NULL) THEN 'visited_no_bp'
+        WHEN (bps.systolic < 140 AND bps.diastolic < 90) THEN 'controlled'
+        ELSE 'uncontrolled'
+        END
+        AS htn_treatment_outcome_in_last_3_months,
+
+    CASE
+        WHEN (visits.months_since_visit >= 2 OR visits.months_since_visit is NULL) THEN 'missed_visit'
+        WHEN (bps.months_since_bp >= 2 OR bps.months_since_bp is NULL) THEN 'visited_no_bp'
+        WHEN (bps.systolic < 140 AND bps.diastolic < 90) THEN 'controlled'
+        ELSE 'uncontrolled'
+        END
+        AS htn_treatment_outcome_in_last_2_months,
+
+    CASE
+        WHEN (visits.quarters_since_visit >= 1 OR visits.quarters_since_visit is NULL) THEN 'missed_visit'
+        WHEN (bps.quarters_since_bp >= 1 OR bps.quarters_since_bp is NULL) THEN 'visited_no_bp'
+        WHEN (bps.systolic < 140 AND bps.diastolic < 90) THEN 'controlled'
+        ELSE 'uncontrolled'
+        END
+        AS htn_treatment_outcome_in_last_quarter
```

Comment:
This is the outcome in _this_ quarter, right? Can we rename to `in_this_quarter`, or `in_quarter`?

Reply:
Ha! Yes, I missed this while renaming it to be similar to the other ones. Will rename to `in_quarter`.